replica/table.cc

```diff
@@ -3950,8 +4024,28 @@ const compaction_manager& compaction_group::get_compaction_manager() const noexc
     return _t.get_compaction_manager();
 }
 
-compaction::table_state& compaction_group::as_table_state() const noexcept {
-    return *_table_state;
+compaction::compaction_group_view& compaction_group::as_view_for_static_sharding() const {
+    return view_for_unrepaired_data();
+}
+
+compaction::compaction_group_view& compaction_group::view_for_unrepaired_data() const {
+    return *_unrepaired_view;
+}
+
+compaction::compaction_group_view& compaction_group::view_for_sstable(const sstables::shared_sstable& sst) const {
+    switch (_repair_sstable_classifier(sst)) {
+        case repair_sstable_classification::unrepaired: return *_unrepaired_view;
+        case repair_sstable_classification::repairing: break; // we need _repairing_view to support this.
```

Comment:
This will cause the node to abort  if we pass a repairing sstable to this function.

Reply:
We could pass a compaction_group_view, but this means we need to check on all the caller site for nullptr.